website/content/en/docs/troubleshooting.md

```diff
@@ -353,6 +353,84 @@ This can be spot checked like shown above, or monitored via the following metric
 operator_status_condition_count{type="ConsistentStateFound",kind="NodeClaim",status="False"}
 ```
 
+### Karpenter Is Unable to Satisfy Topology Spread Constraint
+
+When scheduling pods with TopologySpreadConstraints, Karpenter will attempt to spread the pods across all eligible domains.
+Eligible domains are determined based on the pod's requirements, e.g. node affinity terms.
+However, pod's do not inherit the requirements of compatible NodePools.
+
+For example, consider the following NodePool and Deployment specs:
+
+```yaml
+appVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  template:
+    spec:
+      requirements:
+      - key: topology.kubernetes.io/zone
+        operator: Exists
+---
+appVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: np-zonal-constraint
+  labels:
+    project: zone-specific-project
+spec:
+  template:
+    spec:
+      requirements:
+      - key: topology.kubernetes.io/zone
+        operator: In
+        values: ['us-east-1a', 'us-east-1b']
+      # ...
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inflate
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: inflate
+  template:
+    metadata:
+      labels:
+        app: inflate
+    spec:
+      nodeSelector:
+        project: zone-specific-project
+      topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: topology.kubernetes.io/zone
+          whenUnsatisfiable: DoNotSchedule
+          labelSelector:
+            matchLabels:
+              app: inflate
+```
+
+This cluster has subnets in three availability zones: `us-east-1a`, `us-east-1b`, and `us-east-1c`.
+NodePool `default` can launch instance types in all three zones, but `np-zonal-constraint` is constrained to two.
+Since Karpenter uses the pod's requirements to derive eligible domains, and the pod does not have any zonal constraints, it will consider all known AZs.
+However, Karpenter will only be able to satisfy these constraints for the first two replicas, and will fail to satisfy constraints for all subsequent pods due to the requirements on compatible NodePools.
+In this scenario, Karpenter will fail to scale up further if the TSC is required, or it will always relax the constraint if it is preferred.
+
+In order to prevent these scenarios, you should ensure that all eligible domains for a pod can be provisioned by compatible NodePools, or constrain the pod such that it's eligble domains match those of the NodePools.
```

Comment:
First part of this sentence is pretty confusing IMO but I also can't figure out a better way to write it :D

Reply:
Discussed offline, going to leave this as is. It's definitely difficult to communicate this in a succinct way, I've updated some of the wording above though to use more consistent language which should hopefully make this less confusing.